packages/server/api/src/app/ai/providers/index.ts

```diff
@@ -1,15 +1,24 @@
-import { anthropicProvider } from './anthropic'
-import { googleProvider } from './google'
-import { openaiProvider } from './openai'
-import { replicateProvider } from './replicate'
-import { AIProviderStrategy } from './types'
+import { anthropicProvider, AnthropicProviderConfig } from './anthropic-provider'
+import { googleProvider, GoogleProviderConfig } from './google-provider'
+import { openaiProvider, OpenAIProviderConfig } from './openai-provider'
+import { AIProviderStrategy } from './ai-provider'
+import { openRouterProvider, OpenRouterProviderConfig } from './openrouter-provider'
+import { azureProvider, AzureProviderConfig } from './azure-provider'
 
-export const aiProvidersStrategies: Record<string, AIProviderStrategy> = {
+export type AiProviderConfig = {
+    openai: OpenAIProviderConfig;
+    anthropic: AnthropicProviderConfig;
+    openrouter: OpenRouterProviderConfig;
+    azure: AzureProviderConfig;
+    google: GoogleProviderConfig;
+}
+
+export const aiProviders: Record<string, AIProviderStrategy<any>> = {
```

Comment:
You should not use any and inside the AIProviderStrategy

You should the generic type that's union of all configs (Check the comment about creating typebox object for each provider and infer it type) as the default value.